sagemaker-featurestore/feature_store_kms_key_encryption.ipynb

```diff
@@ -0,0 +1,593 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "policy = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Id\": \"key-policy-feature-store\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Sid\": \"Allow access through Amazon SageMaker Feature Store for all principals in the account that are authorized to use Amazon SageMaker Feature Store\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Principal\": {\n",
+    "                \"AWS\": role\n",
+    "                \n",
+    "            },\n",
+    "            \"Action\": [\n",
+    "                \"kms:Encrypt\",\n",
+    "                \"kms:Decrypt\",\n",
+    "                \"kms:DescribeKey\",\n",
+    "                \"kms:CreateGrant\",\n",
+    "                \"kms:RetireGrant\",\n",
+    "                \"kms:ReEncryptFrom\",\n",
+    "                \"kms:ReEncryptTo\",\n",
+    "                \"kms:GenerateDataKey\",\n",
+    "                \"kms:ListAliases\",\n",
+    "                \"kms:ListGrants\"\n",
+    "            ],\n",
+    "            \"Resource\": [\"*\"],\n",
+    "            \"Condition\": {\n",
+    "                \"StringLike\": {\n",
+    "                    \"kms:ViaService\": \"sagemaker.*.amazonaws.com\"\n",
+    "                }\n",
+    "            }\n",
+    "        },\n",
+    "        {\n",
+    "            \"Sid\": \"Allow administrators to view the CMK and revoke grants\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Principal\": {\n",
+    "                \"AWS\": [role]\n",
+    "            },\n",
+    "            \"Action\": [\n",
+    "                \"kms:Describe*\",\n",
+    "                \"kms:Get*\",\n",
+    "                \"kms:List*\",\n",
+    "                \"kms:RevokeGrant\"\n",
+    "            ],\n",
+    "            \"Resource\": [\"*\"]\n",
+    "        },\n",
+    "        {\n",
+    "            \"Sid\": \"Allow Feature Store to get information about the CMK\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Principal\": {\n",
+    "                \"Service\": \"sagemaker.amazonaws.com\",\n",
+    "                 \"AWS\": [role]\n",
+    "            },\n",
+    "            \"Action\": [\n",
+    "                \"kms:Describe*\",\n",
+    "                \"kms:Get*\",\n",
+    "                \"kms:List*\"\n",
+    "            ],\n",
+    "            \"Resource\": [\"*\"]\n",
+    "        },\n",
+    "        {\n",
+    "            \"Sid\": \"Enable IAM User Permissions\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Principal\": {\n",
+    "                \"AWS\": [role]\n",
+    "            },\n",
+    "            \"Action\": \"kms:*\",\n",
+    "            \"Resource\": [\"*\"]\n",
```

Comment:
Looks like IAM user is not used in this notebook, is this needed?

Reply:
Removed.